go/exoservice/exoservice.go

```diff
@@ -1 +1,77 @@
 package exoservice
+
+import (
+	"fmt"
+
+	"github.com/Originate/exocom/go/exorelay"
+	"github.com/Originate/exocom/go/structs"
+)
+
+// SendHelpers contains helper functions for sending messages
+type SendHelpers struct {
+	Reply func(exorelay.MessageOptions)
+}
+
+// MessageHandler is the function signature for handling a message
+type MessageHandler func(structs.MessagePayload, SendHelpers)
+
+// MessageHandlerMapping is a map from message name to MessageHandler
+type MessageHandlerMapping map[string]MessageHandler
+
+// ExoService is the high level Go API to talk to Exocom
+type ExoService struct {
+	Config          exorelay.Config
+	MessageHandlers MessageHandlerMapping
+	exoRelay        exorelay.ExoRelay
+}
+
+// Connect brings an ExoService instance online
+func (e *ExoService) Connect() error {
+	e.exoRelay = exorelay.ExoRelay{Config: e.Config}
+	err := e.exoRelay.Connect()
+	if err != nil {
+		return err
+	}
+	e.listenForMessages()
+	return nil
+}
+
+// Helpers
+
+func (e *ExoService) listenForMessages() {
+	messageChannel := e.exoRelay.GetMessageChannel()
+	go func() {
+		for {
+			message, ok := <-messageChannel
+			if !ok {
+				break // channel closed
+			}
+			e.receiveMessage(message)
+		}
+	}()
+}
+
+func (e *ExoService) receiveMessage(message structs.Message) {
+	if e.MessageHandlers == nil {
+		return
+	}
+	if e.MessageHandlers[message.Name] == nil {
+		return
+	}
+	e.MessageHandlers[message.Name](message.Payload, SendHelpers{
+		Reply: e.buildSendHelper(message.ID),
+	})
+}
+
+func (e *ExoService) buildSendHelper(responseTo string) func(exorelay.MessageOptions) {
+	return func(options exorelay.MessageOptions) {
+		_, err := e.exoRelay.Send(exorelay.MessageOptions{
+			Name:       options.Name,
+			Payload:    options.Payload,
+			ResponseTo: responseTo,
+		})
+		if err != nil {
+			fmt.Println("Error sending message:", err)
```

Comment:
I'm not sure if quietly logging is what we want to do here. It's probably fine for now, but we should add a TODO or a github issue to update this.

I feel like either the send should be handled by the caller or the service should crash. What do you think?